spec.html

```diff
@@ -35068,6 +35058,109 @@ <h1>Runtime Semantics: AllocateTypedArray ( _constructorName_, _newTarget_, _def
+                1. Set _srcByteIndex_ to _srcByteIndex_ + _srcElementSize_.
+                1. Set _targetByteIndex_ to _targetByteIndex_ + _elementSize_.
+                1. Set _count_ to _count_ - 1.
+            1. Set _O_.[[ViewedArrayBuffer]] to _data_.
+            1. Set _O_.[[ByteLength]] to _byteLength_.
+            1. Set _O_.[[ByteOffset]] to 0.
+            1. Set _O_.[[ArrayLength]] to _elementLength_.
+          </emu-alg>
+        </emu-clause>
+
+        <emu-clause id="sec-initializetypedarrayfromarraybuffer" aoid="InitializeTypedArrayFromArrayBuffer" oldids="sec-typedarray-buffer-byteoffset-length">
+          <h1>InitializeTypedArrayFromArrayBuffer ( _O_, _buffer_, _byteOffset_, _length_ )</h1>
+          <p>The abstract operation InitializeTypedArrayFromArrayBuffer takes arguments _O_ (a TypedArray object), _buffer_ (an ArrayBuffer object), _byteOffset_ (an ECMAScript language value), and _length_ (an ECMAScript language value). It performs the following steps when called:</p>
+          <emu-alg>
+            1. Assert: _O_ is an Object that has a [[TypedArrayName]] internal slot.
+            1. Assert: _buffer_ is an Object that has an [[ArrayBufferData]] internal slot.
+            1. Let _constructorName_ be the String value of _O_.[[TypedArrayName]].
+            1. Let _elementSize_ be the Element Size value specified in <emu-xref href="#table-the-typedarray-constructors"></emu-xref> for _constructorName_.
+            1. Let _offset_ be ? ToIndex(_byteOffset_).
+            1. If _offset_ modulo _elementSize_ &ne; 0, throw a *RangeError* exception.
+            1. If _length_ is not *undefined*, then
+              1. Let _newLength_ be ? ToIndex(_length_).
+            1. If IsDetachedBuffer(_buffer_) is *true*, throw a *TypeError* exception.
+            1. Let _bufferByteLength_ be _buffer_.[[ArrayBufferByteLength]].
+            1. If _length_ is *undefined*, then
+              1. If _bufferByteLength_ modulo _elementSize_ &ne; 0, throw a *RangeError* exception.
+              1. Let _newByteLength_ be _bufferByteLength_ - _offset_.
+              1. If _newByteLength_ &lt; 0, throw a *RangeError* exception.
+            1. Else,
+              1. Let _newByteLength_ be _newLength_ &times; _elementSize_.
+              1. If _offset_ + _newByteLength_ &gt; _bufferByteLength_, throw a *RangeError* exception.
+            1. Set _O_.[[ViewedArrayBuffer]] to _buffer_.
+            1. Set _O_.[[ByteLength]] to _newByteLength_.
+            1. Set _O_.[[ByteOffset]] to _offset_.
+            1. Set _O_.[[ArrayLength]] to _newByteLength_ / _elementSize_.
+          </emu-alg>
+        </emu-clause>
+
+        <emu-clause id="sec-initializetypedarrayfromotherobject" aoid="InitializeTypedArrayFromOtherObject" oldids="sec-typedarray-object">
+          <h1>InitializeTypedArrayFromOtherObject ( _O_, _object_ )</h1>
```

Comment:
how about this name:
```suggestion
          <h1>InitializeTypedArrayFromIterableOrArraylike ( _O_, _object_ )</h1>
```

Reply:
Okay. (But presumably capital "L" in "ArrayLike", as in the rest of the spec.)

Somewhat more cleanly, we could separate the two: `InitializeTypedArrayFromIterable` and `InitializeTypedArrayFromArrayLike`.

Reply:
Splitting sounds good to me.

Reply:
Turns out, for the `Iterable` case, it's simpler for the caller to reduce it to a List, and then pass that.